### PR TITLE
Flite ad serving for amp-ads

### PR DESCRIPTION
**Pull request by andyhills**
Add Flite ads to the amp-ads feature
Issue [#1983](https://github.com/ampproject/amphtml/issues/1983)

### CODE REVIEW

File: ads/flite.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+ export function flite(global, data) {
+     checkData(data, ['guid']);
+     var guid = data.guid, o = global, r = "", m, e = encodeURIComponent, x = 0, url;
+     o.FLITE=o.FLITE || {};
+     o.FLITE.config = o.FLITE.config || {};
+     o.FLITE.config[guid] = o.FLITE.config[guid] || {};
+     o.FLITE.config[guid].cb = Math.random();
+     o.FLITE.config[guid].ts = (+ new Date());
+     try{ r = (top===self && top.location) ? top.location.href : document.referrer || (top.location && top.location.href) || "";}catch(er){x=1}
```
Comment (cramforce) [inline ads/flite.js]: This will never work not throw.

Just make this `r = global.context.location.href`.
`document.referrer` should also work.

Comment (cramforce) [inline ads/flite.js]: You may be interested in `global.context.canonicalUrl`

**Comment by cramforce** [comment]
Looks great. One comment

**Comment by andyhills** [comment]
Fixed relevant comment.

Also, this is being submitted under the company CLA held by Flite

File: ads/flite.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+ export function flite(global, data) {
+     checkData(data, ['guid']);
+     var guid = data.guid, o = global, r = "", m, e = encodeURIComponent, x = 0, url;
+     o.FLITE=o.FLITE || {};
+     o.FLITE.config = o.FLITE.config || {};
+     o.FLITE.config[guid] = o.FLITE.config[guid] || {};
+     o.FLITE.config[guid].cb = Math.random();
+     o.FLITE.config[guid].ts = (+ new Date());
+     try{ r = global.context.location.href; }catch(er){x=1}
```
Comment (cramforce) [inline ads/flite.js]: You can remove the try block

File: ads/flite.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+     try{ m = r.match(new RegExp("[A-Za-z]+:[/][/][A-Za-z0-9.-]+")); } catch(er) {x+=4}
```
Comment (cramforce) [inline ads/flite.js]: Also here.

**Comment by cramforce** [comment]
Looks good. Please squash commits and this can land!

**Comment by andyhills** [comment]
Should be good to go, though I can't see what the conflicts are

**Comment by andyhills** [comment]
Ran into some issue while squashing/rebasing. Will submit separate pull request linked to Intent to Implement issue.

**Pull request closed without merging**